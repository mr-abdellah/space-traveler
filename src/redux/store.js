import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import rocketReducer from './rockets';
import missionsReducer from './missions';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
