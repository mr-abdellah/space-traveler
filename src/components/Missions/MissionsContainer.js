/* eslint-disable jsx-a11y/control-has-associated-label */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Table from 'react-bootstrap/Table';
import { loadMission } from '../../redux/missions';
import Mission from './Mission';
import '../styles/mission.css';

export default function MissionsContainer() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadMission());
  }, []);
  const { missions, pending } = useSelector((state) => state.missions);
  return (
    <div className="mission-container">
      <Table striped bordered hover size="sm">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th />
          </tr>
        </thead>
        <tbody>
          {pending ? (
            <h3>pending</h3>
          ) : (
            missions.map((mission) => (
              <Mission
                key={mission.id}
                id={mission.id}
                name={mission.name}
                description={mission.description}
                reserved={mission.reserved}
              />
            ))
          )}
        </tbody>
      </Table>
    </div>
  );
}
