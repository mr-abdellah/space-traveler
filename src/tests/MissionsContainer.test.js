import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionsContainer from '../components/Missions/MissionsContainer';
import store from '../redux/store';

it('Test missions component:', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <MissionsContainer />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
