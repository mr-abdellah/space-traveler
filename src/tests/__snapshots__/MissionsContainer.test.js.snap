// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test missions component: 1`] = `
<div
  className="mission-container"
>
  <table
    className="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Testing missions component: 1`] = `
<div
  className="mission-container"
>
  <table
    className="table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
